master/buildbot/test/unit/test_util_httpclientservice.py

```diff
@@ -121,6 +121,18 @@ def test_post_auth(self):
                                                                 'Content-Type': 'application/json'
                                                             })
 
+    def test_passed_headers_overwrite_base_headers(self):
```

Comment:
I reviewed that chunk in the other PR. we need tests for the two other backends. (and documentation two document that new feature)